Home work lesson5

PavelChupin opened the pull request:
Добрый день
Реализован ДЗ 5

PavelChupin committed:
Добавлен камент

PavelChupin committed:
Добавлен камент

PavelChupin committed:
Поправлены ошибки ДЗ5

PavelChupin committed:
Переименован входной параметр для метода ДЗ5

PavelChupin committed:
Сделан пример еще с синхронизацией метода по ДЗ5

PavelChupin committed:
Сделан пример еще с синхронизацией метода по ДЗ5

PavelChupin committed:
В ДЗ5 поменяли флаги на boolean

OKrylov commented on src/main/java/homework/lesson5/MainHomeWorkLesson5.java:
```diff
@@ -0,0 +1,10 @@
+    public static void main(String[] args) throws InterruptedException {
+        ArrayClass arrayClass = new ArrayClass();
+        System.out.println("Метод 1 выполнился за " + arrayClass.method1());
+        System.out.println("Метод 2 выполнился за " + arrayClass.method2());
```
Для каждого последующего метода нужно создавать новый экземпляр класса, иначе мы работаем на основе уже вычисленного массива

OKrylov reviewed (approved):
Задание выполнено верно, логика реализована корректно

PavelChupin committed:
Согласно замечания, сделал что бы для первого метода делалась копия целевого массива, для того что бы заново для каждого метода не определять новый объект.
Так же сделал второй вариант с другой реализацией через реализацию интерфейса.
Прошу при наличии времени проверить и дать комментаии.

PavelChupin commented:
Согласно замечания, сделал что бы для первого метода делалась копия целевого массива, для того что бы заново для каждого метода не определять новый объект.

Так же сделал второй вариант с другой реализацией через реализацию интерфейса.
Прошу при наличии времени проверить и дать комментарии.

PavelChupin committed:
Провел полный рефакторинг кода. Убрал повторы в методах.
Варинт второй реализации перенес в метод 4.

PavelChupin commented:
Провел полный рефакторинг кода. Убрал повторы в методах.
Вариант второй реализации перенес в метод 4 основной реализации.
Прошу дать раскрытый каммент по оптимизации кода и основному алгоритму.

OKrylov reviewed (commented):
Все отлично, замечаний не имею
